MatchServer/MMatchRuleGunGame.cpp

```diff
@@ -0,0 +1,134 @@
+#include "stdafx.h"
+#include "MMatchRuleGunGame.h"
+#include "MBlobArray.h"
+
+MMatchRuleGunGame::MMatchRuleGunGame(MMatchStage* pStage) : MMatchRule(pStage)
+{
+}
+
+MMatchRuleGunGame::~MMatchRuleGunGame()
+{
+}
+
+void MMatchRuleGunGame::OnBegin()
+{
+	if (m_pGGSet.size() > 0) {
+		for (auto itor = m_pGGSet.begin(); itor != m_pGGSet.end(); ++itor)
+		{
+			delete *itor;
+		}
+		m_pGGSet.clear();
+	}
+
+	int randSet = rand() % MGetGunGame()->GetSetCount();
```

Comment:
the randset is why i use a map, i make the set the key, and then return a vector of itemsets via MGetGunGame()->GetGGSet(randSet);This allows for specific sets, instead of just 100 items written and then randomly shuffled like most gungames seem to do.  Of course, random_shuffle could still be used on each individual set.